sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -200,24 +202,37 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
             catalog,
             ident,
             parts,
+            query,
             planLater(query),
             props,
             writeOptions,
             orCreate = orCreate) :: Nil
       }
 
     case AppendData(r: DataSourceV2Relation, query, _) =>
-      AppendDataExec(r.table.asWritable, r.options, planLater(query)) :: Nil
```

Comment:
Why we reuse the v2 `CreateTableAsSelectExec` for v1 fallback, but not `AppendDataExec`?

Reply:
because table creation happens after certain operations with side effects in CTAS and RTAS.

Reply:
I think this is because CTAS and RTAS plans don't have a table instance. CTAS and RTAS create a table at runtime and then write to it.

Reply:
This explains why we must reuse `CreateTableAsSelectExec` for v1 fallback, but why can't we reuse `AppendDataExec` for v1 fallback as well?

Reply:
It's just cleaner to have a separation of which code path was used. (Shows up separately in the SQL tab, etc)

Reply:
If end-users look at the SQL tab and see `AppendDataExecV1`, they would expect to see v1 version of CTAS physical plan as well, and may report a bug if they don't see it.

BTW I think there are other ways to implement this feature (users know if v1 fallback is triggered from SQL tab), e.g. we can use SQLMetrics to report it, which can be updated at runtime and support CTAS as well.

Reply:
I don't think most (98%) of Spark users already don't know what the physical nodes stand for, etc. I was thinking of using metrics as well. We just wanted to keep the nodes separate, because the semantics are a bit different for things like `OverwriteByExpression`, where the source's implementation of the Overwrite + Append (v2 behavior) may not be atomic, etc

Reply:
good point on having different physical nodes for different semantics. It seems to me that the semantic of append is the same between v1 and v2. Shall we reuse `AppendDataExec`? Then we can make a good story: If semantic is the same, reuse the v2 plan (CREATE TABLE, CTAS, APPEND). Otherwise, create a new physical node (OVERWRITE).

Reply:
I like the idea to use the same physical plan node when the semantics match, but I'm not sure that it is possible to make strong guarantees about not changing v1 if we do. The problem is that there are multiple v1 plan nodes for the same operation, which could have slightly different behavior.

We could take the time to inspect the v1 implementations and convert, but that adds risk and takes time. It also isn't needed to migrate to v2, and wouldn't speed up the migration, unless I'm missing something. So it probably doesn't provide enough value to make doing it worth while.